--- a/drivers/gpu/drm/v3d/v3d_submit.c	2026-06-10 15:20:58.735807529 +0200
+++ b/drivers/gpu/drm/v3d/v3d_submit.c	2026-06-10 15:21:13.681622798 +0200
@@ -122,24 +122,6 @@
 	v3d_job_free(ref);
 }
 
-static void
-v3d_cpu_job_free(struct kref *ref)
-{
-	struct v3d_cpu_job *job = container_of(ref, struct v3d_cpu_job,
-					       base.refcount);
-
-	v3d_timestamp_query_info_free(&job->timestamp_query,
-				      job->timestamp_query.count);
-
-	v3d_performance_query_info_free(&job->performance_query,
-					job->performance_query.count);
-
-	if (job->indirect_csd.indirect)
-		drm_gem_object_put(job->indirect_csd.indirect);
-
-	v3d_job_free(ref);
-}
-
 void v3d_job_cleanup(struct v3d_job *job)
 {
 	if (!job)
